static/js/subs.js

```diff
@@ -727,20 +727,28 @@ function ajaxSubscribe(stream) {
 
 function ajaxUnsubscribe(sub) {
     // TODO: use stream_id when backend supports it
-    return channel.del({
-        url: "/json/users/me/subscriptions",
-        data: {subscriptions: JSON.stringify([sub.name]) },
-        success: function () {
-            $(".stream_change_property_info").hide();
+    $("#unsubscribe-error").html('');
+    $('#unsubscribe_stream_modal').modal("show");
+    $('#unsubscribe_stream_button').off().on('click', function (e) {
+        e.stopPropagation();
+        e.preventDefault();
+        channel.del({
+            url: "/json/users/me/subscriptions",
+            data: {subscriptions: JSON.stringify([sub.name]) },
+            success: function () {
+                $("#unsubscribe_stream_modal").modal("hide");
             // The rest of the work is done via the unsubscribe event we will get
-        },
-        error: function (xhr) {
-            ui_report.error(i18n.t("Error removing subscription"), xhr,
-                            $(".stream_change_property_info"));
-        },
+            },
+            error: function (xhr) {
+                ui_report.error(i18n.t("Error removing subscription"), xhr,
+                                $(".stream_change_property_info"));
+            },
+        });
```

Comment:
We probably want to show this alert dialog box only if stream is private.

Reply:
I can make this modal appear only when the stream is private but I think it's better to give warning in any case. 
Even though a person can subscribe back to a stream, why not warn him/her for the action. What say?
PS: I can do make this modal work for the private streams, that's not an issue.

Reply:
On it.